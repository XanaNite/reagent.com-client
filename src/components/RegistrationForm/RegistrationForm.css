.register-form{
    margin-top: 50px;
}

.register-input{
    margin-top: 15px;
}

.register-form input[type=text]{
    width: 80%;
    height: 30px;
}

.register-form input[type=password]{
    width: 80%;
    height: 30px;
}

.register-button{
    text-align: center;
}

.register-button button{
    width: 100px;
    background-color: #384CB8;
    border: 0;
    border-radius: 2px;
    height: 40px;
    font-size: 16px;
    margin: 25px;
}

.register-form a{
    text-decoration: none;
    color: whitesmoke;
}