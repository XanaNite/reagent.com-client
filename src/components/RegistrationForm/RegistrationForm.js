import React from 'react'
import './RegistrationForm.css'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'

export default class RegistrationForm extends React.Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }
    
    state = { error: null }

    handleSubmit = ev => {
        ev.preventDefault()
        const {first_name, last_name, user_name, password, agent_email, agent_phone} = ev.target
        this.setState({ error: null })
        console.log(first_name)

        AuthApiService.postAgent({
            first_name: first_name.value,
            last_name: last_name.value,
            user_name: user_name.value,
            password: password.value,
            agent_email: agent_email.value,
            agent_phone: agent_phone.value
        })
        .then(agent => {
            first_name.value = ''
            last_name.value = ''
            user_name.value = ''
            password.value = ''
            agent_email.value = ''
            agent_phone.value = ''
            this.props.onRegistrationSuccess()
        })
        .catch(res => {this.setState({error: res.error})})
    }

    render() {
        const { error } = this.state

        return(
            <form className="register-form" onSubmit={this.handleSubmit}>
                <div role='alert'>{error && <p className='red'>{error}</p>}</div>
                <div className="register-input">
                    <label htmlFor="agent_email">Email address</label>
                    <div>
                        <input type="text" name="agent_email" id="agent_email" />
                    </div>
                </div>
                <div className="register-input">
                    <label htmlFor="password">Password</label>
                    <div>
                        <input type="password" name="password" id="password" />
                    </div>
                </div>
                <div className="register-input">
                    <label htmlFor="user_name">Username</label>
                    <div>
                        <input type="text" name="user_name" id="user_name" />
                    </div>
                </div>
                <div className="register-input">
                    <label htmlFor="first_name">First Name</label>
                    <div>
                        <input type="text" name="first_name" id="first_name" />
                    </div>
                </div>
                <div className="register-input">
                    <label htmlFor="last_name">Last Name</label>
                    <div>
                        <input type="text" name="last_name" id="last_name" />
                    </div>
                </div>
                <div className="register-input">
                    <label htmlFor="agent_phone">Phone number</label>
                    <div>
                        <input type="text" name="agent_phone" id="agent_phone" />
                    </div>
                </div>
                <div className="register-button">
                    <button><a href="dashboard/agentDashboard.html">Next Step</a></button>
                </div>
            </form>
        )
    }
}