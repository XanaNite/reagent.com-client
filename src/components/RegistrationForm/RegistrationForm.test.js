import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';

describe(`RegistrationForm component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the RegistrationForm given props', () => {
        const wrapper = shallow(<RegistrationForm {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
})