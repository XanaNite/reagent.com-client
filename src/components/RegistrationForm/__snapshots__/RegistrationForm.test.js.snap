// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders the RegistrationForm given props 1`] = `
<form
  className="register-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="register-input"
  >
    <label
      htmlFor="agent_email"
    >
      Email address
    </label>
    <div>
      <input
        id="agent_email"
        name="agent_email"
        type="text"
      />
    </div>
  </div>
  <div
    className="register-input"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <div>
      <input
        id="password"
        name="password"
        type="password"
      />
    </div>
  </div>
  <div
    className="register-input"
  >
    <label
      htmlFor="first_name"
    >
      First Name
    </label>
    <div>
      <input
        id="first_name"
        name="first_name"
        type="text"
      />
    </div>
  </div>
  <div
    className="register-input"
  >
    <label
      htmlFor="last_name"
    >
      Last Name
    </label>
    <div>
      <input
        id="last_name"
        name="last_name"
        type="text"
      />
    </div>
  </div>
  <div
    className="register-input"
  >
    <label
      htmlFor="agent_phone"
    >
      Phone number
    </label>
    <div>
      <input
        id="agent_phone"
        name="agent_phone"
        type="text"
      />
    </div>
  </div>
  <div
    className="register-button"
  >
    <button
      type="submit"
    >
      Next Step
    </button>
  </div>
</form>
`;
